### ISSUE TITLE
跑train_shadownet.py 遇见这个问题有谁知道是什么原因么？

### ISSUE THREAD

**Issue by xrf116** [TJCVRS/CRNN_Tensorflow#13]
报错位置在这句代码中cost = tf.reduce_mean(tf.nn.ctc_loss(labels=input_labels, inputs=net_out, sequence_length=25*np.ones(32)))

报错信息：
self._traceback = self._graph._extract_stack()  # pylint: disable=protected-access

InvalidArgumentError (see above for traceback): Saw a non-null label (index >= num_classes - 1) following a null label, batch: 1 num_classes: 37 labels: 
	 [[Node: CTCLoss = CTCLoss[ctc_merge_repeated=true, ignore_longer_outputs_than_inputs=false, preprocess_collapse_repeated=fal

**Comment by TJCVRS** [comment]
@xrf116 应该是你的类别和label对应不上了吧

**Comment by billstark** [comment]
@xrf116 我现在也遇到了同样的问题，我现在想train 的是字符+数字+字母并且改了char_dict.json 和 ord_map.json 请问一下你当时是怎么解决这个问题的？

**Comment by FHainzl** [comment]
I have the same issue as @billstark and would be very interested in any solution regarding training letters+digits+symbols

**Comment by billstark** [comment]
Oh I think I managed to solve that. Here are two things that you need to do after you modified the JSON file:
1. Rerun the script that generates tfrecords.
2. Modify the training file. Change number of classes of “37” to the number of synbals of characters + 1

**Comment by billstark** [comment]
Then you can start your training :)

**Comment by hufangjian** [comment]
你好，问一下，训练的数据集的label 的字符是不是必须要全在char_dict中，我看了一下char_dict有537个字符，中文的应该有5000多个字符，如果训练的中文字符utf-8编码后不在char_dict的537个字符中是不是要自己添加编码？

**Comment by TJCVRS** [comment]
@hufangjian 是的 需要自己添加编码 tools文件下有个工具可以生成字典

**Comment by hufangjian** [comment]
你好，是否是train_text_features.py这个脚本？ 我有个疑问，char_dict.json和ord_map.json文件是根据我生成tfrecord的时候，检索训练集的labels后生成的？还是我提前找一大堆中文文字库生成对应的utf-8编码的？
谢谢！

**Comment by TJCVRS** [comment]
@hufangjian 有一个establish_char_dict.py的工具,　你可以看一下这个工具,　可以生成需要的字典

**Comment by Tian14267** [comment]
请问我现在要训练中文数据，请问我要修改哪里啊？
数据是360W的，你用的那个。我已经生成对应的tfrecords文件了，但是在训练的时候遇到和上面一样的问题。请问我该修改哪些地方？谢谢 @MaybeShewill-CV

**Comment by MaybeShewill-CV** [comment]
@Tian14267 检查global_config.py文件的设置是不是没改:)
